libbeat/common/datetime.go

```diff
@@ -54,14 +70,28 @@ func (t Time) Hash32(h hash.Hash32) error {
 	return err
 }
 
-// ParseTime parses a time in the TsLayout format.
+// ParseTime parses a time in the NanoTsLayout format first, then use millisTsLayout format
 func ParseTime(timespec string) (Time, error) {
-	t, err := time.Parse(TsLayout, timespec)
+	var (
+		t         time.Time
+		err       error
+		tsLayout  string
+		trailZero string
+	)
+
+	for i := 0; i < 3; i++ {
+		trailZero += "000"
+		tsLayout = fmt.Sprintf("%s.%sZ", TsLayout, trailZero)
```

Comment:
`fmt.Sprintf` is a rather expensive operation on the critical path. Given that we have only 3 patterns better create a table with supported layouts. This allows devs to even add more layout types in the future for probing.